framework/include/utils/MooseTypes.h

```diff
@@ -326,6 +326,16 @@ enum LineSearchType
 #endif
 #endif
 };
+
+/**
+ * Type of the matrix-free finite-differencing parameter
+ */
+enum MffdType
+{
+  MFFD_INVALID, /// means not set
```

Comment:
You will need `///<` here so that the comment will be attached to `MFFD_INVALID` by Doxygen. Having it this way, it will be attached to `MFFD_WP`.